### ISSUE TITLE
Printer keeps request line 13 again and again, communication stuck at start of print - Anycubic I3 Mega

### ISSUE THREAD

**Issue by Zolarayze** [foosel/OctoPrint#3448]
**What were you doing?**
I was trying to print a file I just uploaded to the SD card on the OctoPi. It waited correctly for the right temperatures before the print to start.

**What did you expect to happen?**
I was expecting the printer to print the file and not give me this error.

**What happened instead?**
Right at the start of printing, it stops and gives me the error: "Printer keeps requesting line 13 again and again, communicaiton stuck" and switches to offline (target temps back to 0 and no start of the print).

**Did the same happen when running OctoPrint in safe mode?**
In safe mode I get the weird behaviour that was talked about in #2309. First of all it did not heat up the nozzle, target was 195 degrees celcius in OctoPrint, but on the printer it said the target was 0. Next up after pressing "pre-heat pla" on the printer itself, it randomly started to "print" in mid-air (nothing came out of the nozzle, probably because it was cold). I then canceled the print in OctoPrint.
Check the logs below for what happend there.

**Version of OctoPrint**
OctoPrint version 1.3.12

**Operating System running OctoPrint**
OctoPi version 0.17.0

**Printer model & used firmware incl. version**
The printer I have is a 32-bit Anycubic I3 Mega. I saw there were some problems with this specific kind of printer and octoprint on #2309 and upgraded the firmware using the update.zw file and installed both .py scripts that were mentioned there. My firmware version now is V1.4.1RC2/P3. The scripts seem to be loaded as seen in the octoprint.log file.

**Browser and Version of Browser, Operating System running Browser**
Google Chrome  79.0.3945.130, OS: Windows 10 64-bit.

**Link to octoprint.log**
[octoprint.log](https://pastebin.com/euVkk1fD)
[octoprint.log in safe mode](https://pastebin.com/TkM0Sis9)

**Link to contents of terminal tab or serial.log**
[serial.log](https://pastebin.com/Gs56gH5W)
[terminal tab](https://pastebin.com/WxiSnHmt)
[serial.log safe mode](https://pastebin.com/b64PenMD)
[terminal tab safe mode](https://pastebin.com/pt5rLm9A)

If anyone can help me out, it would be greatly appreciated.

**Comment by GitIssueBot** [comment]
Hi @Zolarayze,

It looks like there is some **information missing** from your bug report that will be needed in order to solve the problem. Read the [Contribution Guidelines](https://github.com/foosel/OctoPrint/blob/master/CONTRIBUTING.md) which will provide you with a template to fill out here so that your bug report is ready to be investigated (I promise I'll go away then too!).

If you did not intend to report a bug but wanted to **request a feature or brain storm** about some kind of development, please take special note of the title format to use as described in the [Contribution Guidelines](https://github.com/foosel/OctoPrint/blob/master/CONTRIBUTING.md).

**Please do not abuse the bug tracker as a support forum** - that can be found at [discourse.octoprint.org](https://discourse.octoprint.org). Go there for any kind of issues with network connectivity, webcam functionality, printer detection or any other kind of such support requests or general questions.

Also **make sure you are at the right place** - this is the bug tracker of the official version of OctoPrint, not the Raspberry Pi image OctoPi nor any unbundled third party OctoPrint plugins or unofficial versions. Make sure too that you have **read through the [Frequently Asked Questions](http://faq.octoprint.org)** and searched the [**existing tickets**](https://github.com/foosel/OctoPrint/search?q=&ref=cmdform&type=Issues) for your problem - try multiple search terms please.

I'm marking this one now as needing some more information. Please understand that if you do not provide that information within the next two weeks (until 2020-02-23 15:40 UTC) I'll close this ticket so it doesn't clutter the bug tracker. This is nothing personal, so please just be considerate and help the maintainers solve this problem quickly by following the guidelines linked above. Remember, the less time the devs have to spend running after information on tickets, the more time they have to actually solve problems and add awesome new features. Thank you!

Best regards,
~ Your friendly GitIssueBot

PS: I'm just an automated script, not a human being, so don't expect any replies from me :) Your ticket is read by humans too, I'm just not one of them.

**Comment by foosel** [comment]
Your printers manufacturer decided to ship it with firmware that had been broken by either then or someone else. It sends non standard responses that break the communication protocol and that is the issue here.

There are two plugins available that you can install to fix this but I also strongly recommend to complain to your printer vendor to fix their broken ass firmware. I can't do that for you.

See here for details: https://github.com/foosel/OctoPrint/issues/2309#issuecomment-358713557

**Comment by Zolarayze** [comment]
Hey @foosel thanks for responding, I saw the thread and installed the 2 scripts beforehand (I wrote about this in the main post under printer model and firmware.) . Unfortunately, it didn't fix the issue I am experiencing. It seems a little bit like the problem apos was talking about in #2309, however I tried that solution as well, but without luck.

**Comment by Zolarayze** [comment]
Alright so it seems like the problem was my starting .gcode that I set in Cura.
Figured this out by receiving a different .gcode file that someone send me.
There were 2 commands on a single line.
I also changed Z endpoint to a new line.

`G28 X0 Y0 Z0 ;move X/Y to min endstops`
to
`G28 X0 Y0 ;move X/Y to min endstops
G28 Z0 ;move Z to min endstops`

and 

`G0 Z0.15 G92 E0 ;zero the extruded length`
to
`G0 Z0.15 ;Drop to bed
G92 E0 ;zero the extruded length`

So if anyone else has this problem, check if there are multiple commands on the same line.
